File: jsrc/kernelprolog/fluents/SxxMachine/ListSource.java
package SxxMachine;

import SxxMachine.Nonvar;
import SxxMachine.Copier;
import SxxMachine.Prog;

/**
  Builds an iterator from a list
*/
public class ListSource extends JavaSource {
	public ListSource(Nonvar xs, Prog p) {
		super(Copier.ConsToVector(xs), p);
	}
}

File: src/kernelprolog/fluents/SxxMachine/ListSource.cpp
using namespace std;

#include "ListSource.h"
#include "../../../machine/SxxMachine/Nonvar.h"
#include "../../terms/SxxMachine/Prog.h"

namespace SxxMachine
{

	ListSource::ListSource(Nonvar *xs, Prog *p) : IterableSource(Copier::ConsToVector(xs), p)
	{
	}
}
